Week3/homework/creditCardValidator.js

```diff
@@ -0,0 +1,47 @@
+  'use strict'
+
+
+// The main function.
+function creditCardValidator(creditCardNumber) {
+
+//This function sums the digits of the credit card number.
+ let sum_of_digits = 0;
+ function sum_of_digits_Validator(creditCardNumber) {
+  for (const digits of creditCardNumber) {
+    sum_of_digits += +digits;
+  }
+  return sum_of_digits;
+ }
+ //This function checks whether there is at least two different digits(all of the digits cannot be the same)
+	
+ function allSameDigitsValidator(creditCardNumber) {
+        return /^(\d)\1+$/.test(creditCardNumber.replace(/[^\d]/g,'')); 
+    }
+	
+	
+	/* This function checks whether the count of digits is more than 16 .    and makes sure that all of them are numbers and the last digit is      even ,it calles also the previous functions. 
+	*/
+
+    if (creditCardNumber.toString().length > 16 || creditCardNumber.toString().length < 16) { //This if statement checks  the credit card number whether all.
+        return 'Invalid number:Your credit card namber must be at least 16 digit.'
+    }  else if (isNaN(creditCardNumber)) {
+        return 'Invalid number:Your credit card number should consist of numbers only.'
+    } else if (creditCardNumber.toString()[15] % 2 !== 0) { 
+        return 'Invalid number:The final digit must be even.'
+    } else if (sum_of_digits_Validator(creditCardNumber) < 16) {
+        return 'Invalid number:The sum of all the digits must be greater than 16.'
+    } else if (allSameDigitsValidator(creditCardNumber) === true) {
```

Comment:
Good use of functions to check these validations. Nit: try to keep the naming the same style, e.g. all `camelCase` or `snake_case`.